### ISSUE TITLE
`Unexpected error` when setting up the integration

### ISSUE THREAD

**Issue by Disane87** [meichthys/uptime_kuma#19]
Hi, thank for this project! But I have encountered a pretty weird bug. When I set up the integration, the settings dialog provides me this error:

![image](https://user-images.githubusercontent.com/956158/202792350-216887ae-3d1b-4b24-b810-1808869bb37b.png)

I already checked the credentials, the url (with/without protocol and port etc.) but it won't work. I even disabled the 2FA in Uptime Kuma and changed my password, as mentioned in #10.

I saw an error in the homeassistant logs:
```
Logger: custom_components.uptime_kuma
Source: custom_components/uptime_kuma/config_flow.py:58
Integration: uptime_kuma
First occurred: 09:53:29 (6 occurrences)
Last logged: 19:38:32

could not convert string to float: 'html>'
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/prometheus_client/parser.py", line 114, in _parse_sample
    label_start, label_end = text.index("{"), text.rindex("}")
ValueError: substring not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/uptime_kuma/config_flow.py", line 58, in _validate_input
    await uptime_robot_api.async_get_monitors()
  File "/usr/local/lib/python3.10/site-packages/pyuptimekuma/decorator.py", line 68, in wrapper
    response = UptimeKumaApiResponse.from_prometheus(
  File "/usr/local/lib/python3.10/site-packages/pyuptimekuma/models.py", line 76, in from_prometheus
    for family in parsed:
  File "/usr/local/lib/python3.10/site-packages/prometheus_client/parser.py", line 13, in text_string_to_metric_families
    yield from text_fd_to_metric_families(StringIO.StringIO(text))
  File "/usr/local/lib/python3.10/site-packages/prometheus_client/parser.py", line 209, in text_fd_to_metric_families
    sample = _parse_sample(line)
  File "/usr/local/lib/python3.10/site-packages/prometheus_client/parser.py", line 132, in _parse_sample
    value, timestamp = _parse_value_and_timestamp(text[name_end:])
  File "/usr/local/lib/python3.10/site-packages/prometheus_client/parser.py", line 106, in _parse_value_and_timestamp
    value = float(values[0])
ValueError: could not convert string to float: 'html>'
```

I guess this could be a problem by my dns, because I redirect every call to my subdomains '*.disane.dev' to my local reverse proxy (Nginx proxy manager) which provides the SSL.

All right, I thought to myself. I just take the IP address. and that works. So I guess that problems could result from my dns. But when I browse to my subdomain for uptime kuma, it works flawlessless.

As a DNS I use AdGuard home and some DNS redirects. Do you have any idead how I could solve this?

**Comment by meichthys** [comment]
Can you confirm what version you are installing from HACS?

**Comment by meichthys** [comment]
Maybe double check your `https://` url. The error you copied seems to indicate that you're trying to use `http>` or something similar?

**Comment by Disane87** [comment]
> Can you confirm what version you are installing from HACS?

I added your repo to HACS as custom repo, so I guess this will be v2.



> Maybe double check your `https://` url. The error you copied seems to indicate that you're trying to use `http>` or something similar?

This is not `http>`in the error, it's:
> could not convert string to float: 'html>'

Furrther, I did a cURL to my uptime kuma subdomain  from my HA docker with host network which resulted in this response:
```html
<html>
<head><title>302 Found</title></head>
<body>
<center><h1>302 Found</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

The strange thing is, I had to use the `--insecure` flag for my cURL request because without it, the request failed with `curl: (60) SSL certificate problem: self signed certificate` even when it's a SSL cert from LE.

Did you added your subdomains to the DNS rewrites in AdGuard like this?
![image](https://user-images.githubusercontent.com/956158/202822356-e00bba1a-fd19-42dc-aa70-67b496cebc71.png)

`traceroute` resolved properly to my unraid server, so I guess this isn't really a problem of my DNS. 

But the config of my reverse proxy looks good imho:
![image](https://user-images.githubusercontent.com/956158/202822539-55052ced-d2c8-47b7-a545-52f1ef80c506.png)

So this issue is pretty weird. Especially while it's working with the IP and port I used for my uptime kuma docker with bridged network. So the problem seems to be in the DNS and/or Proxy but the configs seem to ok.

**Comment by meichthys** [comment]
I'm not sure what else to try, but yes, I do have my Adguard rewrites setup the same 👍

**Comment by meichthys** [comment]
I have my uptime kuma accessible publicly, so that is a difference. My guess is still something with dns.

Are you running traceroute from your homeassistant box? That's the machine that needs to be able to resolve uptime kuma.

**Comment by Disane87** [comment]
> I have my uptime kuma accessible publicly, so that is a difference. My guess is still something with dns.

Mine is public too 🙂 


> Are you running traceroute from your homeassistant box? That's the machine that needs to be able to resolve uptime kuma. 

Yeah I tried this from the HA machine.

What I can check is: 
You use the external dep UptimeKuma to connect. I could write something similar only to and check the connection from different machines, so I know if it's the machine or something else. Since I know python this shouldn't be a problem.

**Comment by meichthys** [comment]
Maybe you have your internal network setup differently, but you may want your Access List in Nginxproxymanager to be set to public?

**Comment by Disane87** [comment]
> Maybe you have your internal network setup differently, but you may want your Access List in Nginxproxymanager to be set to public?

Already tried this, nothing changed

**Comment by meichthys** [comment]
Are you using vlans in your setup? 
Or does homeassistant have multiple network adapters?

**Comment by Disane87** [comment]
No, none of this. I don't have any vlan nor does HA has more than one network

**Comment by hans99** [comment]
Same fault for me. Uptime Kuma in Kubernetes behind proxy (Traefik). Works fine in a browser and with curl using http://host:80/metrics
Let me know if I can help with more info!

**Comment by meichthys** [comment]
@jayakornk any ideas here? Seems to be related to the new pyuptimekuma dependency.

**Comment by Disane87** [comment]
> Same fault for me. Uptime Kuma in Kubernetes behind proxy (Traefik). Works fine in a browser and with curl using http://host:80/metrics
> 
> Let me know if I can help with more info!

Thank you for your reporting this! Actually I guess the problem is the reverse proxy and the redirect. 

I hope I can test the dependency for uptime kuma in a blank new project to test how this can be solved.

**Comment by Disane87** [comment]
Had the chance to test this with the example script of https://github.com/jayakornk/pyuptimekuma

This even crashes with the same error:
![image](https://user-images.githubusercontent.com/956158/202931235-e59e18b1-e1e0-4628-8e99-b4fc342a68b6.png)

```
could not convert string to float: 'html>'

During handling of the above exception, another exception occurred:

  File "C:\Users\Disane\OneDrive\Desktop\uptime-kuma-proxy.py", line 17, in main
    response = await uptime_robot_api.async_get_monitors()
  File "C:\Users\Disane\OneDrive\Desktop\uptime-kuma-proxy.py", line 22, in <module>
    loop.run_until_complete(main())
```

I guess this is a problem of the package `pyuptimekuma`

**Comment by meichthys** [comment]
@Disane87 Thanks for setting up the test instance. 
It probably doesn't help you find out what's happening on your end, but your test instance works fine in my HomeAssistant:
![image](https://user-images.githubusercontent.com/10717998/202949674-60eb7586-76e5-4f49-b840-a2c0cedeb797.png)
![image](https://user-images.githubusercontent.com/10717998/202949696-ca19e902-7260-485e-9265-6bcb91e6245c.png)
🤷‍♂️

**Comment by Disane87** [comment]
Thank you for the test! In my networks this fails, so the only thing which is differing from internal and external call is the AdGuard DNS.

**Comment by meichthys** [comment]
Yes it does seem like it is dns related. Either in Adguard or in homeassistant.

**Comment by dougmaitelli** [comment]
I am having the exact same error too:

`could not convert string to float: 'html>'
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/prometheus_client/parser.py", line 114, in _parse_sample
    label_start, label_end = text.index("{"), text.rindex("}")
ValueError: substring not found
`

**Comment by dannytsang** [comment]
I also get the same error:
```
2022-11-26 22:39:42.487 ERROR (MainThread) [custom_components.uptime_kuma] Request exception for '192.168.1.4:3001/metrics' with - 192.168.1.4:3001/metrics
```

Unt-icked verify SSL and it was using http locally.

When I use external URL with SSL, it went through fine.

**Comment by meichthys** [comment]
@dougmaitelli Please check your dns configuration.

**Comment by Disane87** [comment]
Since others have this problem too: 

I've fixed my issue. It was a problem of DNS and my infrastructure.

All my incoming traffic of port 443 is forwarded to my NPM which routes per domain binding to the appropiate services. This worked perfectly but I saw, the pings from the internal network to my domains (which are set in NPM) went out and than over my public IP back in and that was totally crap.

I thought the dns rewrites would fix this, but they made it much worse because they pointed to my internal unraid which listens to port 80/443 and on this was the self signed cert and not the `/metrics` endpoint. Instead this the response was my unraid ui.

IHope this helps others :)
